### DISCUSSION TITLE
Migrate custom post type with classic metaboxes to the Gutenberg UI

### DISCUSSION THREAD

**Discussion by mrleemon** [WordPress/developer-blog-content#121] [Topic Ideas]
I'd like to suggest and idea for a topic for the Developer blog. Many devs who use WordPress as a CMS make a heavy use of custom post types with lots of associated meta. With the classic interface this was super easy to achieve thanks to meta boxes, but now the Gutenberg UI has complicated things a lot for us. I know that the `PluginSidebar` and `PluginDocumentSettingPanel` components exist, but they are only appropriate for custom post types with few meta fields or where meta info is not super important, not for custom post types where meta data is the main content, such as the one in the The Events Calendar plugin:

![screenshot-5](https://github.com/WordPress/developer-blog-content/assets/2060659/abc57078-c060-4df1-bf65-eadd69f503f8)
 
The ideal scenario would be to be able to replicate the Start/End Date & Time and Location fields in the main content section of the Gutenberg editor, but I have yet to see any tutorial dealing with this.

**Comment by justintadlock** [comment]
We do have one published post that dives into this: https://developer.wordpress.org/news/2023/03/creating-a-custom-block-that-stores-post-meta/

But, we could definitely use more content and examples around dealing with custom meta.

**Reply by mrleemon** [reply]
I'd like some examples on how to use a custom block with meta fields in the `template` parameter in `register_post_type` function. And an example showing how to restrict the use of this custom block only in a specific post type and how to hide it from the inserter.
